sound/soc/sof/sof-audio.c

```diff
@@ -94,6 +102,9 @@ static int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swid
 				 &reply, sizeof(reply));
 	if (ret < 0) {
 		dev_err(sdev->dev, "error: failed to free widget %s\n", swidget->widget->name);
+		mutex_lock(&swidget->use_count_mutex);
+		swidget->use_count++;
```

Comment:
If we get a error, this widget will never be released ?

Reply:
I think this is ok. The widget remains loaded, so it makes sense to keep the use_count reflecting real status. If the hw is powered down, the use_count is reset in any case.